services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```diff
@@ -55,18 +65,49 @@
     private Queue executorQueue;
 
     /**
+     * To avoid problems with passing detached entities,
+     * load related entities during cloning
+     */
+    protected AnalysisContext cloneExecution(AnalysisContext originalContext)
+    {
+        Set<Package> allPackages = new HashSet<>();
+
+        allPackages.addAll(originalContext.getIncludePackages());
+        allPackages.addAll(originalContext.getExcludePackages());
+
+        List<Long> packageIds = allPackages.stream().map(aPackage -> aPackage.getId())
+                .collect(Collectors.toList());
+
+        AnalysisContext clone = originalContext.clone();
```

Comment:
@OndraZizka @jsight I'm trying to pass full `AnalysisContext` entity to execute method (instead of just passing `id`) but it is fairly more complicated - I'm getting exceptions about using Detached state entities. 

It is because `AnalysisContext` now doesn't come from persistence context so its related entities either. I need to load all related entities from persistence context (so they are in Managed state). Is there any simpler way, then reading ids of all related entities and getting them by JPA query?